.github/workflows/ci.yaml

```diff
@@ -0,0 +1,95 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main   # push commit to the main branch
+    tags:
+      - 'v2*'  # push tag starting with "v2" to the main branch
+  pull_request:
+    branches:
+      - main   # pull request to the main branch
+  workflow_dispatch:   # allow manual triggering
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build-linux:
+    name: CI py${{ matrix.python-version }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.7', '3.8', '3.9', '3.10']
+      max-parallel: 5
+    env:
+      python_version: ${{ matrix.python-version }}
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Deploy Information
+        if: ${{ startsWith(github.ref, 'refs/tags') && env.python_version == '3.7' }}
+        run: |
+          echo "The HTML NeXus User Manual will be pushed to"
+          echo " https://github.com/nexusformat/definitions/tree/gh-pages"
+          echo "The HTML NeXus User Manual will be deployed on"
+          echo " https://nexusformat.github.io/definitions/"
+
+      - name: Install Requirements
+        run: |
+          python3 -m pip install --upgrade pip setuptools
+          make install
+          python3 -m pip list
+
+      - name: Check Code Style
+        run: |
+          make style
+
+      - name: Run Tests
+        run: |
+          make test
+
+      - name: Install LaTeX
+        run: |
+          sudo apt-get update -y && \
+          sudo apt-get install -y \
+            latexmk \
+            texlive-latex-recommended \
+            texlive-latex-extra \
+            texlive-fonts-recommended
+
+      - name: Generate build files
+        run: |
+          make prepare
+
+      - name: Build Impatient Guid
+        run: |
+          make impatient-guide
+          ls -lAFgh build/impatient-guide/build/html/index.html
+          ls -lAFgh build/impatient-guide/build/latex/NXImpatient.pdf
+
+      - name: Build User Manual
+        run: |
+          make pdf
+          make html
+          ls -lAFgh build/manual/build/html/index.html
+          ls -lAFgh build/manual/build/latex/nexus.pdf
+
+      - name: Build and Commit the User Manual
```

Comment:
consider PR #1147 here

Reply:
Mean what? CNAME should be created by the prepare stage?

Reply:
Ok I added the generation of the file to the prepare stage of the manual